File: Practica2/linea.py
from punto import Punto

class Linea:
    def __init__(self, p1: Punto, p2: Punto):
        self.p1 = p1
        self.p2 = p2
    
    def __str__(self):
        return f"Linea(p1={self.p1}, p2={self.p2})"
    
    def dibujaLinea(self):
        print(f"Dibujando línea desde {self.p1} hasta {self.p2}")

# Ejemplo de uso:
if __name__ == "__main__":
    p1 = Punto(0, 0)
    p2 = Punto(3, 4)
    linea = Linea(p1, p2)
    print(linea)
    linea.dibujaLinea()

File: Practica2/linea.java
public class Linea {
    private Punto p1;
    private Punto p2;

    public Linea(Punto p1, Punto p2) {
        this.p1 = p1;
        this.p2 = p2;
    }

    @Override
    public String toString() {
        return "Linea(p1=" + p1 + ", p2=" + p2 + ")";
    }

    public void dibujaLinea() {
        System.out.println("Dibujando línea desde " + p1 + " hasta " + p2);
    }

    // Ejemplo de uso en un main (opcional)
    public static void main(String[] args) {
        Punto punto1 = new Punto(0, 0);
        Punto punto2 = new Punto(3, 4);
        Linea linea = new Linea(punto1, punto2);
        System.out.println(linea);
        linea.dibujaLinea();
    }
}
